pkg/crd/generator/testData/pkg/apis/fun/v1alpha1/toy_types.go

```diff
@@ -70,6 +70,7 @@ type ToyStatus struct {
 // +kubebuilder:printcolumn:name="toy",type="string",JSONPath=".status.conditions[?(@.type==\"Ready\")].status",description="descr1",format="date",priority=3
 // +kubebuilder:printcolumn:name="abc",type="integer",JSONPath="status",description="descr2",format="int32",priority=1
 // +kubebuilder:printcolumn:name="service",type="string",JSONPath=".status.conditions.ready",description="descr3",format="byte",priority=2
+// +kubebuilder:resource:path=services,shortName=ty
```

Comment:
We should add another type where `resource` annotation is not added to cover both the scenarios because existing projects do not have resource annotation and I am afraid this change will break `crd generation` for them.

Reply:
> We should add another type where `resource` annotation is not added to cover both the scenarios.

Could you please explain this a bit more in detail?

Reply:
add a test case without the resource annotation, just to make sure we do the sane thing if the annotation isn't present.

Reply:
The changes included in the code are handling all the conditions.